presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -286,6 +288,36 @@ public static double stLength(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return geometry.getEsriGeometry().calculateLength2D();
     }
 
+    @SqlNullable
+    @Description("Returns a float between 0 and 1 representing the location of the closest point on the LineString to the given Point, as a fraction of total 2d line length.")
+    @ScalarFunction("line_locate_point")
+    @SqlType(DOUBLE)
+    public static Double lineLocatePoint(@SqlType(GEOMETRY_TYPE_NAME) Slice lineSlice, @SqlType(GEOMETRY_TYPE_NAME) Slice pointSlice)
+    {
+        if (lineSlice == null || pointSlice == null) {
+            return null;
+        }
+
+        Geometry line = JtsGeometrySerde.deserialize(lineSlice);
+        Geometry point = JtsGeometrySerde.deserialize(pointSlice);
+
+        if (line.isEmpty() || point.isEmpty()) {
```

Comment:
Did you test these semantics in posgresql?

Reply:
I didn't, but returning `null` when computation doesn't make sense is consistent with the rest of the functions.

Reply:
```
postgres=# select ST_LineLocatePoint(ST_GeomFromText('LINESTRING EMPTY'), ST_GeomFromText('POINT EMPTY')) IS NULL;
ERROR:  getPoint2D_const_p: point offset out of range
```

Yeah, just null is probably better.